types.go

```diff
@@ -111,14 +119,19 @@ type Problem struct {
 	// Func evaluates the objective function at the given location. Func
 	// must not modify x.
 	Func func(x []float64) float64
+
 	// Grad evaluates the gradient at x and stores the result in-place in grad.
 	// Grad must not modify x.
 	Grad func(x []float64, grad []float64)
+
```

Comment:
Also not for this PR, but I think we should change the input order. The gonum rule is that "answer receivers" are the first inputs.